src/DeviceInterface/InterfacePrototype.js

```diff
@@ -702,6 +706,31 @@ class InterfacePrototype {
     openManageCreditCards () {}
     openManageIdentities () {}
 
+    /**
+     * This method takes availableInputTypes and credentials (from the native side),
+     * and updates autofill UI and data.
+     * @param {import('../Settings.js').AvailableInputTypes} availableInputTypes
+     * @param {CredentialsObject[]} credentials
+     */
+    updateAutofillInputs (availableInputTypes, credentials) {
+        // Update local settings and data
+        this.settings.setAvailableInputTypes(availableInputTypes)
+        this.storeLocalCredentials(credentials)
+
+        // rerender the tooltip
+        this.uiController?.updateItems(credentials)
+
+        if (!this.globalConfig.isTopFrame) {
+            // If the tooltip is open on an autofill type that's not available, close it
+            const currentInputSubtype = getSubtypeFromType(this.getCurrentInputType())
+            if (!availableInputTypes.credentials?.[currentInputSubtype]) {
+                this.removeTooltip()
+            }
+            // Redecorate fields according to the new types
+            this.scanner.forms.forEach(form => form.recategorizeAllInputs())
```

Comment:
Should this rather be `form.redecorateAllInputs`? So we avoid re-scanning the form.

Reply:
This is just a refactor I did for third party provider, when I was trying to re-use it. Actually this part is totally unrelated now (original approach it was related), I will just put it back in the older state.